server/gui/options_parser.py

```diff
@@ -0,0 +1,24 @@
+import shlex
+
+import click
+
+from server.cli.launch import common_args
+
+
+@click.command()
+@common_args
+def cli(**kwargs):
+    pass
+
+
+def parse_opt_string(opts):
```

Comment:
how much of this function is using private (click-internal) interfaces, vs. public API?   Just thinking about maintaining this in the long term....     Short term, could protect by adding some unit tests.

Reply:
`add_to_parser` (line 18) is the only truly internal function. Technically OptionParser is an internal class, but it's included in their documentation. 

I'll add some unit tests. Long term, click may be the wrong option if we want to keep this pattern with the gui. Reading some of the issues on click's github implies they are deliberately against adding the flexibility I'd need to do this the right way.